src/olympia/addons/tasks.py

```diff
@@ -191,23 +193,50 @@ def recreate_theme_previews(addon_ids, **kw):
         '[%s@%s] Recreating previews for themes starting at id: %s...'
         % (len(addon_ids), recreate_theme_previews.rate_limit, addon_ids[0])
     )
-    addons = Addon.objects.filter(pk__in=addon_ids).no_transforms()
+    version_ids = Addon.objects.filter(pk__in=addon_ids).values_list('_current_version')
+    versions = Version.objects.filter(pk__in=version_ids)
     only_missing = kw.get('only_missing', False)
 
-    for addon in addons:
-        version = addon.current_version
-        if not version:
-            continue
+    renders = {
+        render['full']: {
+            'thumb_size': render['thumbnail'],
+            'thumb_format': render['thumbnail_format'],
+        }
+        for render in amo.THEME_PREVIEW_RENDERINGS.values()
+    }
+
+    for version in versions:
         try:
             if only_missing:
```

Comment:
Just so that we're clear: you only need to do the search for which things to recreate when `only_missing` is `True` because in the other case, we recreate _everything_ using `generate_static_theme_preview()` ?

Reply:
yes.  It's for when we change the preview svg and need to rebuild everything.  (It's perhaps a use-case we don't need any longer, but I left it there)